src/kademlia/get_peers.cpp

```diff
@@ -72,10 +72,12 @@ void get_peers_observer::reply(msg const& m)
 			char const* end = peers + n.list_at(0).string_length();
 
 #ifndef TORRENT_DISABLE_LOGGING
+			auto logger = get_observer();
 			bdecode_node const id = r.dict_find_string("id");
```

Comment:
technically, this lookup could be moved inside the ``should_log()`` check as well